python/Ganga/GPIDev/Lib/GangaList/GangaList.py

```diff
@@ -421,7 +434,10 @@ def __str__(self):
         return self.toString()
 
     def append(self, obj, my_filter=True):
-        self._list.append(self.strip_proxy(obj, my_filter))
+        elem = self.strip_proxy(obj, my_filter)
+        if isType(elem, GangaObject):
+            stripProxy(elem)._setParent(stripProxy(self)._getParent())
```

Comment:
Yes, you've understood correctly.

I'm not a fan of this but this is the 'old style' behaviour and a lot of code is written to just naively expect that the ```_getParent``` method from a subjob will return the parent job.
I've modified the master attribute in the Job class to be more intelligent in future and I'm checking for any outstanding ```_getParent``` calls which I can replace with either ```job.master``` or ```job.getJobObject``` which would allow me to properly define the parent of elements in a GangaList as the GangaList itself.

I would rather do this work on a separate pull request however as I see it quickly growing as a task and at the moment I'm seeing problems with submitting simple jobs which is all I want to address here.